src/fancyline/tty/vt100.cr

```diff
@@ -10,11 +30,32 @@ class Fancyline
         super()
       end
 
-      # FIXME: Properly get the terminal dimensions, don't rely on environment
-      #        variables that are outdated at the time of reading them.
+      # Currently always raises when not compiled with `x86_64` and `darwin` flags
+      def winsize
+        {% if flag?(:x86_64) && flag?(:darwin) %}      
+          winsize = uninitialized LibC::Winsize
+          LibC.ioctl(0, LibC::TIOCGWINSZ, pointerof(winsize))
+          winsize
+        {% else %}
+          raise "#winsize not supported on your platform"
```

Comment:
If `#winsize` was part of a shard meant to tell you the size of your terminal I may have used an exception-on-error too.  But it's basically an internal function to `fancyline`, so we can save us from having to do the `raise+rescue` dance.  This is also faster, as raising exceptions is somewhat costly.

Also exceptions are mainly for pants-on-fire errors which the code module itself may not be able to recover from (Or not "usually").  An exception can then tell the caller (The user of fancyline in this case) that something is on fire.  The developer can then choose to fix&retry it, or to tell the user that the printer is on fire.

On error, just return `nil`, and check in the getter methods below if it's `nil`, and if it is, fall back. You can use `Object#try` to great effect here: `may_return_nil.try(&.the_field) || fallback_value`. This works fine as long neither `nil` nor `false` (nor `Pointer.null` if being pedantic) are expected result values.  As they're not, it works here.